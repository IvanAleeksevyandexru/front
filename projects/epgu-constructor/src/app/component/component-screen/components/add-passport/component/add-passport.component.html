<epgu-constructor-screen-pad class="screen-pad-body">
  <section class="add-passport-screen">
    <div class="add-passport-screen__body">
      <epgu-constructor-passport
        [attrs]="$any(data)"
      >
      </epgu-constructor-passport>
    </div>
  </section>
</epgu-constructor-screen-pad>
