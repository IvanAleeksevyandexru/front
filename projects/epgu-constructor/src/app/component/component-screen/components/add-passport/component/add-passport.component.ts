import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { ComponentBase } from '../../../../../screen/screen.types';

@Component({
  selector: 'epgu-constructor-add-passport',
  templateUrl: './add-passport.component.html',
  styleUrls: ['./add-passport.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AddPassportComponent {
  @Input() data: ComponentBase;
}
