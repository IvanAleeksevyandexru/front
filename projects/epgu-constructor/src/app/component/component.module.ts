import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { SharedModule } from '../shared/shared.module';


/**
 * Домен компанент. Сдесь храниться всё что связанно со компанентами и их обязками.
 *
 * TODO: Сюда будем переносить компаненты из домена скринов
 */
@NgModule({
  imports: [
    CommonModule,
    SharedModule
  ],
})
export class ComponentModule {}
