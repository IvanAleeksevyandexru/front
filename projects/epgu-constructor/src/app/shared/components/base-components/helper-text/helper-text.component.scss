@import "~@epgu/epgu-constructor-ui-kit/src/styles/core";

.helper-text {
  @include typ-plain-small;
  color: $color-text-helper;

  &-lg {
    @include typ-plain-text;
  }
}
