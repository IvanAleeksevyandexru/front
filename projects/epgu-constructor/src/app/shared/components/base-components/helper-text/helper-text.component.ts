import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'epgu-constructor-helper-text',
  templateUrl: './helper-text.component.html',
  styleUrls: ['./helper-text.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class HelperTextComponent {
  @Input() size: 'lg';
}
