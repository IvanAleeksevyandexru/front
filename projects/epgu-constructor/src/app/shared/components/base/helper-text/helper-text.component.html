<span class="helper-text" [class.helper-text-lg]="size === 'lg'"><ng-content></ng-content></span>
