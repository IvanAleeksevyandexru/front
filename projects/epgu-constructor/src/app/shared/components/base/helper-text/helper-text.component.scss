@import "../../../../../styles/core";

.helper-text {
  @include typ-plain-small;
  color: $color-text-helper;
}
