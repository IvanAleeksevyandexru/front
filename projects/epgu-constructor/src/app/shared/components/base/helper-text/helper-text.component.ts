import { Component, Input } from '@angular/core';

@Component({
  selector: 'epgu-constructor-helper-text',
  templateUrl: './helper-text.component.html',
  styleUrls: ['./helper-text.component.scss'],
})
export class HelperTextComponent {
  @Input() size: 'lg';
}
