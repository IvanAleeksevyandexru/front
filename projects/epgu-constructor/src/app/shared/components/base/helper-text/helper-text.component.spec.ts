import { Component } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { HelperTextComponent } from './helper-text.component';

const helperTextMock = 'Awesome useful helper text.';

@Component({
  template: `<epgu-constructor-helper-text>${helperTextMock}</epgu-constructor-helper-text>`
})
class WrapperTestComponent {}

describe('HelperTextComponent', () => {
  let component: WrapperTestComponent;
  let fixture: ComponentFixture<WrapperTestComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [ WrapperTestComponent, HelperTextComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(WrapperTestComponent);
    component = fixture.componentInstance;
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('component should render passed text', () => {
    const helperText = fixture.debugElement
      .query(By.css('.helper-text')).nativeElement.textContent;
    expect(helperText).toEqual(helperTextMock);
  });

  it('should created', () => {
    expect(component).toBeTruthy;
  });
});
