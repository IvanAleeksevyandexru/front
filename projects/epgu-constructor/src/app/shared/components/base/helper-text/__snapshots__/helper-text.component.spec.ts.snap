// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelperTextComponent should match snapshot 1`] = `
<ng-component>
  <epgu-constructor-helper-text>
    <span class="helper-text">
      Awesome useful helper text.
    </span>
  </epgu-constructor-helper-text>
</ng-component>
`;
